snowflake/src/main/java/org/apache/iceberg/snowflake/SnowflakeCatalog.java

```diff
@@ -0,0 +1,254 @@
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace)
+      throws NoSuchNamespaceException {
+    SnowflakeIdentifier id = NamespaceHelpers.toSnowflakeIdentifier(namespace);
+    boolean namespaceExists;
+    switch (id.type()) {
+      case DATABASE:
+        namespaceExists = snowflakeClient.databaseExists(id);
+        break;
+      case SCHEMA:
+        namespaceExists = snowflakeClient.schemaExists(id);
+        break;
+      default:
+        throw new IllegalArgumentException(
+            String.format(
+                "loadNamespaceMetadat must be at either DATABASE or SCHEMA level; got %s from namespace %s",
+                id, namespace));
+    }
+    if (namespaceExists) {
+      return ImmutableMap.of();
+    } else {
+      throw new NoSuchNamespaceException(
+          "Namespace '%s' with snowflake identifier '%s' doesn't exist", namespace, id);
+    }
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
+    throw new UnsupportedOperationException(
+        "SnowflakeCatalog does not currently support dropNamespace");
+  }
+
+  @Override
+  public boolean setProperties(Namespace namespace, Map<String, String> properties) {
+    throw new UnsupportedOperationException(
+        "SnowflakeCatalog does not currently support setProperties");
+  }
+
+  @Override
+  public boolean removeProperties(Namespace namespace, Set<String> properties) {
+    throw new UnsupportedOperationException(
+        "SnowflakeCatalog does not currently support removeProperties");
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    String fileIOImpl = DEFAULT_FILE_IO_IMPL;
+    if (catalogProperties.containsKey(CatalogProperties.FILE_IO_IMPL)) {
+      fileIOImpl = catalogProperties.get(CatalogProperties.FILE_IO_IMPL);
+    }
+
+    // Initialize a fresh FileIO for each TableOperations created, because some FileIO
+    // implementations such as S3FileIO can become bound to a single S3 bucket. Additionally,
+    // FileIO implementations often support only a finite set of one or more URI schemes (i.e.
+    // S3FileIO only supports s3/s3a/s3n, and even ResolvingFileIO only supports the combination
+    // of schemes registered for S3FileIO and HadoopFileIO). Individual catalogs may need to
+    // support tables across different cloud/storage providers with disjoint FileIO implementations.
+    FileIO fileIO = fileIOFactory.newFileIO(fileIOImpl, catalogProperties, conf);
+    closeableGroup.addCloseable(fileIO);
+    return new SnowflakeTableOperations(snowflakeClient, fileIO, catalogName, tableIdentifier);
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+    throw new UnsupportedOperationException(
+        "SnowflakeCatalog does not currently support defaultWarehouseLocation");
+  }
+
+  @Override
+  public void setConf(Object conf) {
+    this.conf = conf;
+  }
+
+  public Object getConf() {
```

Comment:
I don't believe this is necessary since we only dynamically set the configuraiton when dynamically loading a catalog implementation.

Reply:
Removed